R/create_credentials.R

```diff
@@ -120,6 +126,39 @@ create_smtp_creds_key <- function(id,
   # Determine whether the keyring package can be used
   validate_keyring_capable()
 
+  # Stop function if `id` provided is not of the right type
+  if (!(inherits(id, "character") | inherits(id, "numeric") | inherits(id, "integer"))) {
+    stop("The provided `id` value must be a single character or numeric value.",
+         call. = TRUE)
+  }
+
+  # Stop function if the `id` value isn't of length 1
+  if (length(id) != 1) {
+    stop("Only a vector of length 1 should be used for the `id` value.",
+         call. = FALSE)
+  }
+
+  # Stop function if the `id` is an empty string or NA
+  if (id == "" || is.na(id)) {
+    stop("The value for `id` should not be `NA` or an empty string.",
+         call. = FALSE)
+  }
+
+  if (is.character(id) & grepl("-", id)) {
+    stop("Hyphens are not allowed as characters for an `id` value",
+         call. = FALSE)
+  }
+
+  # Determine whether the key already exists, stop if it does
+  creds_tbl <- get_keyring_creds_table()
+
+  if (id %in% creds_tbl$id) {
+    stop("The specified `id` corresponds to a credential key already in the key-value store:\n",
+         "* Use a different `id` value with `create_smtp_creds_key()`, or\n",
+         "* Delete the existing key with `delete_credential_key(id = \"", id,"\") and try again",
+         call. = FALSE)
```

Comment:
It'd be nice to have an argument to this function that allows overwrite if it already exists.

Reply:
Included the `overwrite` (default is `FALSE`). An associated message has been changed to inform users of this option.